src/Audacia.Core/EnumMember.cs

```diff
@@ -285,6 +288,39 @@ public static class EnumMember
             throw new ArgumentException($"Requested value '{value}' was not found.");
         }
 
+        /// <summary>
+        /// Attempts to parse the enum as a string and return the relevant member.
+        /// </summary>
+        /// <param name="type">Type of Enum.</param>
+        /// <param name="value">Value to convert.</param>
+        /// <param name="enumValue">Return value of the enum if parsed.</param>
+        /// <returns>The parsed enum.</returns>
+        /// <exception cref="OverflowException"> Throws when the enum member is out of range.</exception>
+        public static bool TryParseEnumAsString(Type type, string value, out object? enumValue)
+        {
+            bool enumDefined;
+
+            try
+            {
+                // Match by number value or field name
+                enumValue = Enum.Parse(type, value, ignoreCase: true);
+                enumDefined = Enum.IsDefined(type, enumValue);
+            }
+            catch (ArgumentException)
+            {
+                // Ignore enum value not found error
+                enumValue = null;
+                enumDefined = false;
+            }
+            
+            if (!enumDefined && enumValue != null)
+            {
+                throw new OverflowException($"Value '{value}' is outside the range of '{(type == null ? "the provided enum type" : type.Name)}'.");
```

Comment:
Could this return false rather than throwing? The fact that the method is a try parse one implies it's not going to throw an exception given a bad input.

In fact this could possible be simplified to return the value of `isDefined`.

Reply:
I swapped it to do exactly this before PR but then a previously written unit test checking for this exception started failing so I added it back in, I can possibly remove the test?

Reply:
Hmmm, ok. That's fair. Couple of things it would be good to do:
1. Make the `TryParseEnumAsString` method private (as I _think_ it's a brand new method? If so I'd keep it private for now).
2. Add a couple of comments in this method explaining that the logic (in terms of exception throwing/handling) is here for backwards compatibility reasons.